shell/platform/darwin/macos/framework/Source/FlutterView.mm

```diff
@@ -84,6 +86,17 @@ - (void)reshaped {
                             }];
 }
 
+- (NSColor*)backgroundColor {
+  return _backgroundColor ? _backgroundColor : [NSColor blackColor];
+}
+
+- (void)paintContainingLayerBackgroundColor {
```

Comment:
Actually, I think the view has layer created anyway (I'm not entirely sure what the hierarchy is right). So maybe just override / forward`setBackgroundColor` on `FlutterView` to the layer.

Reply:
The `_containingLayer` is provided to the `FlutterSurfaceManager` during initialization. Could you explain why forwarding `setBackgroundColor` to the Surface Manager would be better than at the `FlutterView` layer?
 https://github.com/flutter/engine/blob/ecf2f85cab61c0b64f12d77a89afb74cdd8f96ff/shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm#L26

Reply:
Also, if `setBackgroundColor` propagates down to `FlutterSurfaceManager`, I think that we're going to have to provide overrides going all the way up to `FlutterViewController` anyway since the `FlutterView` or `FlutterSurfaceManager` isn't exported as a part of `FlutterMacOS`. But please correct me if I'm not seeing something here.

Reply:
I meant setting the background color on `self.layer` (where self is the FlutterView.) But not just do it in constructor, do it every time user calls [FlutterView setBackgroundColor]. Not doing it in FlutterSurfaceManager.

i.e.

```objc
-(void)setBackgroundColor:(NSColor*)color {
   self.layer.backgroundColor = color;
}
```

And the in contructor ensure `wantsLayer=YES` and set layer color to black.

Reply:
👍 If I understand you correctly, the background color will always be black during creation. The client of the API can only override the background color after instantiation.

Reply:
Sounds good.

Also, do you have any idea about how we can expose `setBackgroundColor` to a Flutter project? IIRC, `FlutterView` is not available via `FlutterMacOS`.

Reply:
Good point. Maybe the setter needs to be on FlutterViewController then? At least for the time being. I'm not sure what the longer term plan with multi-view support is. That might result in exposing `FlutterView` in public api. cc @goderbauer

Reply:
@knopp thanks for the help. This PR should have the changes requested. 

I'm holding off on adding adding `setBackgroundColor` to `FlutterViewController` while awaiting @goderbauer's response. If we can expose `FlutterView` as a public api, it won't make sense to add it to the `FlutterViewController`.

Reply:
I spoke with @goderbauer offline and it looks like we don't have concrete plans for exposing `FlutterView` in the near future. 

To land this, we will have to propagate `setBackgroundColor` to the `FlutterViewController`.